### ISSUE TITLE
UserMsg: Not Present on Client 58

### ISSUE THREAD

**Issue by metita** [Garey27/hitbox_fixer#25]
Basically tested your latest alpha release with live players, some players weren't able to play because they got kicked due to Host_Error : UserMsg: Not Present on Client 58.

![image](https://github.com/user-attachments/assets/d8c233d3-d25a-458f-864b-67c93bfa9df1)

@dystopm saw that 58 was targetting svc_sendcvarvalue2, which makes sense that some clients does not have that message in their clients.

![image](https://github.com/user-attachments/assets/50cd80a6-aaf1-4b1c-a8b4-0395a5cc7d04)

Shoutout to @s1lentq for some little insight about that maybe QueryClientCvar2 usage was for debug purpose rather than being used in production.

**Comment by Garey27** [comment]
On which version of the client did this error occur?

**Comment by metita** [comment]
Exe version 1.6.3.7 (cstrike)
Exe build: 10:56:59 Oct 3 2005 (3266)

**Comment by zik32** [comment]
need update client 2005 to 2009

require min:
Protocol version 48
Exe version 1.1.2.6 (cstrike)
Exe build: 15:34:33 Jun 15 2009 (4554)

**Comment by metita** [comment]
Yep, its due to the usage of queryclientcvar2 which was implemented in later builds of the game AFAIK, but I am not really into having to force people to update their clients just for the usage of a module, make it universal regardless on where they are.

**Comment by metita** [comment]
@Garey27 any chance for this to get a look soon? I would like to try this on a populated environment but some clients cannot join the server

**Comment by kalil-camera** [comment]
older clients can never be supported at the same time, there is just too much work to support that 1% nonsteam nonupdated clients.

force older clients to update, its 2025. pay 10 bucks to use steam.

**Comment by metita** [comment]
> older clients can never be supported at the same time, there is just too much work to support that 1% nonsteam nonupdated clients.
> 
> force older clients to update, its 2025. pay 10 bucks to use steam.

Your statement makes no sense.

**Comment by Garey27** [comment]
@metita @zik32 Just use 1.1.5 version, 2.x.x are in alpha stage and i forgot that old clients don't had queryclientcvar2.

**Comment by metita** [comment]
> [@metita](https://github.com/metita) [@zik32](https://github.com/zik32) Just use 1.1.5 version, 2.x.x are in alpha stage and i forgot that old clients don't had queryclientcvar2.

Don't worry, we will wait for 2.x.x versions without queryclientcvar2, thanks !

**Comment by lexzor** [comment]
Hello, just ran in the same problem and i discovered this module caused the error in photo for old clients.

In my case, i'm running a HNS server with limitation of 250 FPS. Some players does not have a performant PC and they use some old ass clients which gives them FPS boost through some .cfg files and some low poly models.

It would be nice to have a workaround for this, maybe a cvar.

**Comment by dystopm** [comment]
> Hello, just ran in the same problem and i discovered this module caused the error in photo for old clients.
> 
> In my case, i'm running a HNS server with limitation of 250 FPS. Some players does not have a performant PC and they use some old ass clients which gives them FPS boost through some .cfg files and some low poly models.
> 
> It would be nice to have a workaround for this, maybe a cvar.

It is already, but in the experimental branch. You can check code (in branch) here: https://github.com/Garey27/hitbox_fixer/blob/2f737e531c4d9880f92f3ab6bbb2e050b0d9b619/src/main.cpp#L319-L322
